immich-app/immich
# chore(ml): added locustfile

## Description

This PR adds support for [Locust](https://locust.io/), a performance measurement tool. The benefit of this is being able to easily quantify and visualize ML performance. In turn, this makes it easier to understand the performance implications of a change, identify bottlenecks, and find the most appropriate defaults. Besides `locustfile.py` itself, it also adds a small Bash script to deploy the app locally, start Locust, and terminate the app once done. 

## Testing

This PR has been tested to work with the provided poetry.lock dependencies
1. `cd machine_learning`
2. poetry install (install poetry first if not already installed)
3. chmod +x load_test.sh
4. ./load_test.sh
5. Open `localhost:8089` to see the web UI
6. Click `Start swarming`
